CONTRIBUTING.md

```diff
@@ -40,177 +264,782 @@ since someone else can often do this if you include a complete example. We
+  and `dune`).
+
+- A plugin tutorial is located in
+  [`doc/plugin_tutorial`][plugin-tutorial].
+
+- The Coq [wiki][] contains additional developer resources.
+
+#### Building Coq ####
+
+The list of dependencies can be found in the first section of the
+[`INSTALL`](INSTALL) file.
+
+Today, the recommended method for building Coq is to use `dune`.  Run
+`make -f Makefile.dune` to get help on the various available targets,
+Additional documentation can be found in
+[`dev/doc/build-system.dune.md`][dev-doc-dune], and in [the official
+Dune documentation][dune-doc].
+
+The legacy make-based system is still available.  If you wish to use
+it, you need to start by running `./configure -profile devel`.  Most
+of the available targets are not documented, so you will need to ask.
+
+#### Continuous integration ####
+
+Continuous integration (CI) testing is key in ensuring that the
+`master` branch is kept in a well-functioning state at all times, and
+that no accidental compatibility breakages are introduced.  Our CI is
+quite extensive since it includes testing many external projects, some
+of them taking more than an hour to compile.  However, you can get
+partial results much more quickly (when our CI is not overloaded).
+
+The main documentation resources on our CI are:
+
+- the [README for users, i.e. plugin and library authors][CI-README-users];
+- the [README for developers, and contributors][CI-README-developers];
+- the README of the [user-overlays][] directory.
+
+Preparing an overlay is a step that everyone goes through at some
+point.  All you need to know to prepare an overlay manually is in the
+README in the [user-overlays][] directory.  You might
+want to use some additional tooling such as the `make ci-*` targets of
+`Makefile.ci`, the Nix support for getting the dependencies of the
+external projects (see the README in [`dev/ci/nix`](dev-ci-nix), and
+the (undocumented so far) `dev/tools/create_overlays.sh` script.
+
+More work is to be done on understanding how each developer proceeds
+to prepare overlays, and propose a simplified and documented
+procedure.
+
+We also have a benchmarking infrastructure, which is documented [on
+the wiki][jenkins-doc].
+
+#### Code owners, issue and pull request templates ####
+
+These files can be found in the [`.github`](.github) directory.  The
+templates are particularly useful to remind contributors what
+information we need for them, and, in the case of PRs, to update the
+documentation, changelog, and test-suite when relevant.
+
+GitHub now supports setting up multiple issue templates, and we could
+use this to define distinct requirements for various kind of bugs,
+enhancement and feature requests.
+
+#### Style guide ####
+
+There exists an [old style guide][old-style-guide] whose content is
+still mostly relevant.  We don't use a code formatter at the current
+time, and we are reluctant to formatting-only commits, or commits
```

Comment:
```suggestion
time, and we are reluctant to merge formatting-only commits, or commits
```

You could just say "we are reluctant to merge changes to parts of the code that are unchanged aside from formatting".  That implicitly includes "formatting-only commits".